### PR TITLE
test(robot): expand PVC more than storage maximum size

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#6633

#### What this PR does / why we need it:

Add a robot test case to verify that a PVC cannot be expanded beyond the storage maximum size.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

## Release Notes

- **New Features**
  - Enhanced persistent volume claim creation with flexible configuration options.
  - New keyword for verifying persistent volume claim requested size over time.
  - New methods for volume size retrieval and maximum disk storage checks.
  - Added functionality for expanding workloads and persistent volume claims with improved size management.
  - New utility functions for managing custom resources and size conversions.
  - Enhanced logging and error handling in AWS operations.
  - New test suite for validating persistent volume claim behavior, including checks for expansion limits.

- **Bug Fixes**
  - Improved error handling and logging in backup operations.
  - Enhanced error handling and logging in Rest class methods.
  - Refined error handling for annotation setting in utility functions.

- **Style**
  - Updated phrasing in test cases for clarity regarding volume expansion actions.

### CODE REVIEW

**Comment by c3y1huang** [comment]
@coderabbitai review

**Comment by c3y1huang** [comment]
https://github.com/coderabbitai review

File: e2e/tests/regression/test_persistentvolumeclaim.robot [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        When Expand deployment 0 volume more than storage maximum size should fail
+        Then Assert volume size of deployment 0 remains 2GiB for at least 5 seconds
+        And Assert persistentvolumeclaim 0 requested size remains 2GiB for at least 5 seconds
+        And Check deployment 0 data in file data.txt is intact
+
+        When Expand deployment 0 volume by 1 GiB
+        And Assert persistentvolumeclaim 0 requested size remains 3GiB for at least 5 seconds
+        And Assert volume size of deployment 0 remains 3GiB for at least 5 seconds
+        And Check deployment 0 data in file data.txt is intact
+    END
```
Comment (c3y1huang) [inline e2e/tests/regression/test_persistentvolumeclaim.robot:44]: cc @WebberHuang1118 for behavior validation.

File: e2e/tests/regression/test_persistentvolumeclaim.robot [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    And Create deployment 0 with persistentvolumeclaim 0
+    And Write 10 MB data to file data.txt in deployment 0
+
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        When Expand deployment 0 volume more than storage maximum size should fail
+        Then Assert volume size of deployment 0 remains 2GiB for at least 5 seconds
+        And Assert persistentvolumeclaim 0 requested size remains 2GiB for at least 5 seconds
+        And Check deployment 0 data in file data.txt is intact
+
+        When Expand deployment 0 volume by 1 GiB
```
Comment (roger-ryao) [inline e2e/tests/regression/test_persistentvolumeclaim.robot:40]: For this step, I feel a bit confused when executing it. The console output shows the following:
Screenshot_20241209_163327

From reading the test story, I initially thought we needed to expand the original `storage_size` from `2GiB` to `1GB`. Can this part be modified?

**Review by roger-ryao — CHANGES_REQUESTED** [review]

File: e2e/tests/regression/test_persistentvolumeclaim.robot (hunk 2 above)
Comment (c3y1huang) [reply e2e/tests/regression/test_persistentvolumeclaim.robot]: Updated. Thank you!

File: e2e/libs/keywords/persistentvolumeclaim_keywords.py [hunk 3]
```diff
@@ -20,9 +20,9 @@ def cleanup_persistentvolumeclaims(self):
         for claim in claims.items:
             self.delete_persistentvolumeclaim(claim.metadata.name)
 
-    def create_persistentvolumeclaim(self, name, volume_type="RWO", sc_name="longhorn"):
+    def create_persistentvolumeclaim(self, name, volume_type="RWO", sc_name="longhorn", storage_size="3Gi"):
```
Comment (yangchiu) [inline e2e/libs/keywords/persistentvolumeclaim_keywords.py:23]: By running test cases based on this PR, I got many test cases failed with error: `ValueError: Invalid size format: 3Gi`:

https://ci.longhorn.io/job/private/job/longhorn-e2e-test/2036/console

**Review by yangchiu — CHANGES_REQUESTED** [review]

**Review by roger-ryao — APPROVED** [review]
LGTM

File: e2e/tests/regression/test_persistentvolumeclaim.robot [hunk 4]
```diff
@@ -0,0 +1,44 @@
+*** Settings ***
+Documentation    PersistentVolumeClaim Test Cases
+
+Test Tags    regression
+
```
Comment (yangchiu) [inline e2e/tests/regression/test_persistentvolumeclaim.robot:5]: Importing `variables.resource` can help prevent teardown failure when powering nodes on:

https://ci.longhorn.io/job/private/job/longhorn-e2e-test/2067/robot/report/log.html

**Review by yangchiu — CHANGES_REQUESTED** [review]

File: e2e/tests/regression/test_persistentvolumeclaim.robot (hunk 4 above)
Comment (c3y1huang) [reply e2e/tests/regression/test_persistentvolumeclaim.robot:5]: Updated. Thank you!

**Review by yangchiu — APPROVED** [review]
LGTM
